components/LocationList.js

```diff
@@ -1,18 +1,27 @@
-import Card from "./Card";
+import LocationCard from "./LocationCard";
 import styled from "styled-components";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import { StyledList } from "./StyledList";
 
-const StyledList = styled.ul`
-  display: flex;
-  flex-direction: column;
-  align-items: center;
-  padding-left: 0;
+const StyledLink = styled(Link)`
+  text-decoration: none;
 `;
 
 export default function LocationList({ locations }) {
+  const router = useRouter();
+
   return (
     <StyledList>
-      {locations.map((location, index) => (
-        <Card location={location} key={index} />
+      {console.log(locations)}
+      {locations.map((location, id) => (
```

Comment:
Try to have an unique ID, instead of using the map index, this can cause issues later on. You could consider using from your locations array an unique identifier, or use a library like your nanoid which you already installed